pycbc/workflow/core.py

```diff
@@ -198,7 +198,10 @@ def __init__(self, cp, name,
             raise TypeError("Failed to find %s executable " 
                             "at %s" % (name, exe_path))
         
-        self.add_pfn(exe_path)
+        if exe_path.startswith('gsiftp://'):
+            self.add_pfn(exe_path,site='pycbc-code')
```

Comment:
Okay, it sounds like you mostly agree with me Alex. Although I take the point that the site= variable can be used to "prefer" one entry over another (although I might want to prefer one non-local entry over another ... maybe because the file is "closer" or "quicker" to get at and I'm not sure you could do that here!) . I'm still missing why site="nonlocal" is not the right thing to do though?